cli/azd/pkg/project/service_target_containerapp.go

```diff
@@ -85,19 +88,29 @@ func (at *containerAppTarget) Deploy(
 				return
 			}
 
-			// Login, tag & push container image to ACR
-			containerDeployTask := at.containerHelper.Deploy(ctx, serviceConfig, packageOutput, targetResource, true)
-			syncProgress(task, containerDeployTask.Progress())
-
-			_, err := containerDeployTask.Await()
-			if err != nil {
-				task.SetError(err)
-				return
+			if serviceConfig.Docker.RemoteBuild {
+				err := at.remoteBuildHelper.RunRemoteBuild(ctx, serviceConfig, targetResource, func(s string) {
+					task.SetProgress(NewServiceProgress(s))
+				})
```

Comment:
I feel like we should be consistent with the way we update progress until we move over to a new model? 🤷‍♂️

Reply:
Just checking - you mean turning this into our TaskWithProgress pattern and flowing the progress as we do when one Task has to call the other?  Happy to do that.